elastic/kibana
# [Fleet] Rotate key pair API and Service

## Summary

Allows a fleet user to reset saved object encrypted key if/when Kibana is compromised via an API call. 

### How to test it

- Run Kibana locally and check for key pair to be generated and assigned to the agent policies.
Use the Kibana dev console and run
```json5
GET .kibana/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "type": "fleet-message-signing-keys"
          }
        }
      ]
    }
  }
}
```
- Request a `POST` to the API route `api/fleet/message_signing_service/rotate_key_pair?acknowledge=true` 
- Check that the key pair has been rotated in SO.

### Outdated

- and in the existing agent policies in the `.kibana` index using the query 
above.
- Verify that the policies also have the new public key as `agent.protection.signing_key` in the `.fleet-policies` index
```json5
GET .fleet-policies/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "policy_id": "<policy_id>"
          }
        }
      ]
    }
  }
}
```

Note: Viewing agent policies using the Fleet->Agent Policies->View Policy at `http://<kibanaURL>/app/fleet/policies` should also show the same info as the above query.
